src/components/editor/metaform-editor-rightDrawer-feature.tsx

```diff
@@ -0,0 +1,193 @@
+  fieldIndex,
+  pendingForm,
+  setPendingForm
+}) => {
+  const [ metaformSectionTitle, setMetaFormSectionTitle] = React.useState<string | undefined>("");
+  /**
+   * Set values of Confidition field, switch and get selected component name
+   * 
+   */
+  const setVisiblityComponentValues = () => {
+    if (fieldIndex === undefined && sectionIndex !== undefined) {
+      setMetaFormSectionTitle(pendingForm.sections![sectionIndex].title ?? "");
+    }
+  };
+
+  useEffect(() => {
+    setVisiblityComponentValues();
+  }, [fieldIndex, sectionIndex]);
+
+  /**
+   * Updates metaform field
+   *
+   * @param newMetaformField new metaform field
+   */
+  const updateFormField = (newMetaformField: MetaformField) => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, newMetaformField);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Updates metaform section
+   *
+   * @param newMetaformSection new metaform section
+   */
+  const updateFormSection = (newMetaformSection: MetaformSection) => {
+    if (sectionIndex === undefined) {
+      return;
+    }
+
+    const section = pendingForm.sections?.[sectionIndex];
+    if (!section) {
+      return;
+    }
+
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.splice(sectionIndex, 1, newMetaformSection);
+    });
+
+    setPendingForm(updatedForm);
+  };
+
+  /**
+  * Renders fields category title
+  * 
+  * @param title Category title
+  */
+  const renderFieldsCategoryTitle = (title: string) => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      return (
+        <Typography variant="subtitle1" style={{ width: "100%" }}>
+          {title}
+        </Typography>
+
+      );
+    }
+  };
+
+  /**
+   * Render feature components
+   * 
+   */
+  const renderFeatures = () => {
+    if (sectionIndex === undefined && fieldIndex === undefined) {
+      return (
+        <Typography>{ strings.draftEditorScreen.editor.visibility.selectVisibilityInfo }</Typography>
+      );
+    }
+
+    if (sectionIndex !== undefined && fieldIndex === undefined) {
+      const section = pendingForm.sections![sectionIndex];
+      return (
+        <TextField
+          fullWidth
+          value={ section.title }
+          label={ strings.draftEditorScreen.editor.features.sectionTitle }
+          onChange={ event => updateFormSection({
+            ...section,
+            title: event.target.value
+          }) }
+        />
+      );
+    }
+
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      const field = pendingForm.sections![sectionIndex].fields![fieldIndex];
+      if (field === undefined) {
+        return null;
+      }
+
+      if (field !== undefined) {
```

Comment:
Isn't this if statement redundant, since it is already being checked in the earlier if statement that _field_ is not undefined?